docs/pages/07_service.adoc

```diff
@@ -0,0 +1,955 @@
+
+The actor object can grant permission to exports containing services. After granting permission to an export, the user can call all the services in the export.
+
+To authorize actors to use services, use `grants` in the actor specification.
+
+*Syntax:*
+
+----
+actor <name> ([<entity> <alias>]) [grants <export>, ...]
+    [realm "<realm>"]
+    [identity <identity>];
+----
+
+where the `grants` keyword is followed by a comma-separated list of exports.
+
+*Example:*
+
+----
+entity Person {
+    field String email ;
+}
+
+transfer PersonTransfer(Person person) {
+    field String email = salesPerson.email ;
+}
+
+export ManagerExport {
+    service static PersonTransfer create() {
+        // implement this body in Java
+    }
+}
+
+actor Manager grants ManagerExport;
+----
+
+The example above allows the manager to create a new person. To grant permissions to external users, we first need to define entities, transport objects and exports. The Manager actor in this application has no realm or identity, meaning this actor is an anonymous actor. Anonymous actors are publicly available, anyone who knows the URL of the Manager actor interface can invoke the granted create service. This way, the Manager actor is given access to the ManagerExport's create service, with which the user can create a PersonTransfer.
+
+*Permission graph*
+
+// https://en.wikipedia.org/wiki/Graph-based_access_control
+
+JUDO's security mechanism is based on a logical permission graph, where the entry point for external users to the application is the actor. Calling the services granted to the actor the user can access transfer objects. Once the user has access to a transfer object, they can call the services granted by the transfer object (and can access further transfer objects).
+
+The primary benefit of this mechanism is that it provides a direct mapping between the user's permissions and the transfer objects and services they can access. This makes it easier to manage permissions and control access to the application.
+
+The example below shows how a transfer object can grant services. Users who received the PersonTransfer object can call the delete service.  This service can only be called by the users who have access to the PersonTransfer object.
+
+*Syntax:*
+
+----
+transfer <name> (<entity> <alias>) [grants <export>, ...] [{
+    [member] ...
+}]
+----
+
+*Example:*
+
+----
+transfer PersonTransfer(Person person) grants PersonExport {
+    field String email = salesPerson.email ;
+}
+
+export PersonExport(Person person) {
+    service void delete() {
+        // implement this body in Java
+    }
+}
+----
+
+The complete authorization mechanism can be defined by carefully provisioning services to actors and transfer objects. Always be aware of that if a transfer object has already been received by an external user, then they are authorized to call the services provided by the transfer object.
+
```

Comment:
I think the actor definition cant contain grants clause. Instead of the export need to have a grants caluse that contains a list of actors. The couse of this change the code will be more readable for the human eyes, and esier to find that who have to acces a service or transfer objects. Since, in this case I can reach the item, that I looking for step by step in the hierarchy tree. The original case I have to start the searching (stepping) from the top and the bottom in the same time. It is not problem for a machine, but a human won't like this mechanism.

If my comment isn't understandable, create a medell with 3 actors, 3 exports with 2 services, and 3 mapped transfer objects. And every actor will be able to access only 2 exports and use a similar pattern between exports and tranfer objects. And try to answer the next question. Which actors can access one of defined transfer objects.